src/doc/src/reference/build-scripts.md

```diff
@@ -123,6 +132,11 @@ one detailed below.
   scripts.
 
 
+[^link]: since some linkers like [`gcc`][gcc-link-options] require that dynamic
+libraries are specified last in certain cases.
```

Comment:
I'm not aware of such requirements.
Where they come from?

The ordering requirement is relevant to all kinds of libraries, both static and dynamic, especially given that rustc uses `--as-needed` for dynamic libraries by default.

Reply:
@tmfink I was wondering if you can follow up here?  Perhaps the footnote can just be removed?

Reply:
Sorry--missed this comment originally. Compilers/linkers like `gcc/`ld` impose these requirements. On NetBSD at least, `--as-needed` was passed but "wrong" order of link arguments still caused linking to fail.

Here's the relevant snippet from the [GCC manual](https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html):

> -llibrary
> -l library
> 
> ...
> 
>  It makes a difference where in the command you write this option; the linker searches and processes libraries and object files in the order they are specified. Thus, ‘foo.o -lz bar.o’ searches library ‘z’ after file foo.o but before bar.o. If bar.o refers to functions in ‘z’, those functions may not be loaded.

And from the [`ld` manual](https://sourceware.org/binutils/docs/ld/Options.html#Options):

> -l namespec
> --library=namespec
> 
> Add the archive or object file specified by namespec to the list of files to link. This option may be used any number of times. If namespec is of the form :filename, ld will search the library path for a file called filename, otherwise it will search the library path for a file called libnamespec.a.
> 
> ...
> 
> The linker will search an archive only once, at the location where it is specified on the command line. If the archive defines a symbol which was undefined in some object which appeared before the archive on the command line, the linker will include the appropriate file(s) from the archive. However, an undefined symbol in an object appearing later on the command line will not cause the linker to search the archive again.
> 
> See the -( option for a way to force the linker to search archives multiple times.
> 
> You may list the same archive multiple times on the command line.
> 
> This type of archive searching is standard for Unix linkers. However, if you are using ld on AIX, note that it is different from the behaviour of the AIX linker. 


The linker (which would get args in the order passed to the compiler) only looked for symbols in a library if there that symbol had already been "seen" in previous argument. Thus, in practice, you usually want to pass the `-lfoo` link arguments last.

Reply:
I think the concern for this footnote is that the order of arguments matters for all kinds of libraries (and linker arguments in general), not just dynamic libraries.  There's nothing specific about dynamic libraries needing to be last.  Of course the order matters and you may need them to be last, but I think the note above makes it sufficiently clear that the order matters.

Reply:
I removed the footnote